vdp/pipeline/v1beta/component_definition.proto

```diff
@@ -0,0 +1,388 @@
+  // A list of component definition resources.
+  repeated ComponentDefinition component_definitions = 1;
+  // Next page token.
+  string next_page_token = 2;
+  // Total number of connector definitions.
+  int32 total_size = 3;
+}
+
+// ListConnectorDefinitionsRequest represents a request to list connector
+// definitions.
+message ListConnectorDefinitionsRequest {
+  // The maximum number of connector definitions to return. If this parameter
+  // is unspecified, at most 10 definitions will be returned. The cap value for
+  // this parameter is 100 (i.e. any value above that will be coerced to 100).
+  optional int32 page_size = 1 [(google.api.field_behavior) = OPTIONAL];
+  // Page token.
+  optional string page_token = 2 [(google.api.field_behavior) = OPTIONAL];
+  // field 3 is reserved for the "view" property of ConnectorDefinition.View
+  // type, removed in favour of ComponentDefinition.View.
+  reserved 3;
+  // Filter can hold an [AIP-160](https://google.aip.dev/160)-compliant filter
+  // expression.
+  // - Example: `create_time>timestamp("2000-06-19T23:31:08.657Z")`.
+  optional string filter = 4 [(google.api.field_behavior) = OPTIONAL];
+  // View allows clients to specify the desired resource view in the response.
+  optional ComponentDefinition.View view = 5 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// ListConnectorDefinitionsResponse contains a list of connector definitions.
+message ListConnectorDefinitionsResponse {
+  // A list of connector definition resources.
+  repeated ConnectorDefinition connector_definitions = 1;
+  // Next page token.
+  string next_page_token = 2;
+  // Total number of connector definitions.
+  int32 total_size = 3;
+}
+
+// GetConnectorDefinitionRequest represents a request to fetch the details of a
+// connector definition.
+message GetConnectorDefinitionRequest {
+  // The resource name of the connector definition, which allows its access by ID.
+  // - Format: `connector-definitions/{id}`.
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {type: "api.instill.tech/ConnectorDefinition"},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+field_configuration: {path_param_name: "connector_definition_name"}
+    }
+  ];
+  // field 2 is reserved for the "view" property of ConnectorDefinition.View
+  // type, removed in favour of ComponentDefinition.View.
+  reserved 2;
+  // View allows clients to specify the desired resource view in the response.
+  optional ComponentDefinition.View view = 3 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// GetConnectorDefinitionResponse contains the requested connector definition.
+message GetConnectorDefinitionResponse {
+  // The connector definition resource.
+  ConnectorDefinition connector_definition = 1;
+}
+
+// LookUpConnectorDefinitionAdminRequest represents a request by an admin to
+// query a connector definition by its UID.
+message LookUpConnectorDefinitionAdminRequest {
```

Comment:
@donch1989 are the `LookUp[Connector|Operator]DefinitionAdmin` endpoints in use? Can we remove them?

Reply:
They are not in use as well. I think we can remove them.

Reply:
Done 🧹